### DISCUSSION TITLE
How to extract diagrams in research papers?

### DISCUSSION THREAD

**Discussion by Show-han** [jsvine/pdfplumber#628] [Ask for help with specific PDFs]
I'd like to extract diagrams, which include several curves and rects, but not images, in research papers.
This kinds of diagrams are as follows
![image](https://user-images.githubusercontent.com/57925186/159108087-29ff5d9e-6da2-435e-bdbe-aebc0d243ad5.png)
and
![image](https://user-images.githubusercontent.com/57925186/159108134-193995ca-cada-443e-9d9b-b3f761be30b1.png)
and
![image](https://user-images.githubusercontent.com/57925186/159108166-e58456a7-34a4-4828-bffe-11908a19e873.png)
If I treated them as images, they can't be extracted correctly. So how can I extract them to a file? If possible, I want to reserve the entire information of these diagrams so that I can regenerate them in another PDF file.

**Comment by jsvine** [comment]
Hi @Show-han, and thanks for your interest in this library. *Identifying* the full extent of a figure could be a complex / non-trivial task, given that different PDFs may have very different approaches to displaying / rendering them. But once identified, you could *extract* the visual representation of the diagrams via `my_page.crop((x0, top, x1, bottom)).to_image().save("my/diagram/path.png")`.

Does that answer your question? If not, could you provide more specifics?
